backend-implicaction/src/test/java/com/dynonuggets/refonteimplicaction/controller/JobsPostingControllerTest.java

```diff
@@ -126,4 +126,61 @@ void getJobByIdWithoutJwtShouldBeForbidden() throws Exception {
 
         verify(jobPostingService, times(0)).getJobById(anyLong());
     }
+
+    @Test
+    @WithMockUser
+    void archiveJobShouldChangeStatus() throws Exception {
+        //given
+        JobPostingDto givenDto = JobPostingDto.builder()
+                .id(1L)
+                .isArchive(false)
+                .build();
+
+        JobPostingDto expectedDto = JobPostingDto.builder()
+                .id(1L)
+                .isArchive(true)
+                .build();
+
+        given(jobPostingService.toggleArchiveJobPosting(anyLong())).willReturn(expectedDto);
+
+        //when
+        final ResultActions resultActions = mvc.perform(patch(JOBS_BASE_URI + ARCHIVE_JOB_URI, givenDto.getId()).contentType(APPLICATION_JSON));
+
+        //then
+        resultActions.andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is((givenDto.getId().intValue()))))
+                .andExpect(jsonPath("$.archive", is(expectedDto.isArchive())));
```

Comment:
ça serait peut être plus logique de vérifier que `$.archive` est l'inverse de `givenDto.isArchive()` nan ? et comme du coup tu n'utilises plus `expectedDto`, tu pourrais supprimer cette variable.

On gagnerait en :
 * lisibilité (moins de code à lire)
 * et en logique, parce qu'on veut vérifier au final que `archive` est l'opposé de ce qu'on lui a donné à la base
```suggestion
                .andExpect(jsonPath("$.archive", is(!givenDto.isArchive())));
```